File: eCinema/eCinema.Services/Database/Obavijest.cs
using System;
using System.Collections.Generic;

namespace eCinema.Services.Database
{
    public partial class Obavijest
    {
        public int ObavijestId { get; set; }
        public string? Naslov { get; set; }
        public string? Podnaslov { get; set; }
        public string? Sadrzaj { get; set; }
        public string? Slika { get; set; }
        public DateTime? DatumKreiranja { get; set; }
        public int? ObavijestKategorijaId { get; set; }
        public int? KorisnikId { get; set; }

        public virtual Korisnik? Korisnik { get; set; }
        public virtual ObavijestKategorija? ObavijestKategorija { get; set; }
    }
}

File: ecinema_mobile/lib/models/obavijest.dart
import 'package:ecinema_mobile/models/models.dart';
import 'package:json_annotation/json_annotation.dart';

part 'obavijest.g.dart';

@JsonSerializable()
class Obavijest {
  int obavijestId;
  String naslov;
  String podnaslov;
  String sadrzaj;
  String? slika;
  DateTime datumKreiranja;
  int? obavijestKategorijaId;
  KategorijaObavijest? obavijestKategorija;
  int? korisnikId;
  Korisnik korisnik;

  Obavijest(
      {required this.obavijestId,
      required this.naslov,
      required this.podnaslov,
      required this.sadrzaj,
      this.slika,
      required this.datumKreiranja,
      this.obavijestKategorijaId,
      this.obavijestKategorija,
      this.korisnikId,
      required this.korisnik});

  factory Obavijest.fromJson(Map<String, dynamic> json) =>
      _$ObavijestFromJson(json);
  Map<String, dynamic> toJson() => _$ObavijestToJson(this);
}
